pkg/plugin/pr-sanitizer/event_handler.go

```diff
@@ -82,25 +71,53 @@ func (gh *GitHubLabelsEventsHandler) HandleIssueCommentEvent(log log.Logger, com
 	return err
 }
 
-func (gh *GitHubLabelsEventsHandler) checkTitleAndSetStatus(log log.Logger, pullRequest *gogh.PullRequest) error {
-	change := ghservice.NewRepositoryChangeForPR(pullRequest)
-	statusContext := github.StatusContext{BotName: gh.BotName, PluginName: ProwPluginName}
-	statusService := status.NewStatusService(gh.Client, log, change, statusContext)
+func (gh *GitHubPRSanitizerEventsHandler) checkTitleDescriptionAndSetStatus(log log.Logger, pr *gogh.PullRequest) error {
+	change := ghservice.NewRepositoryChangeForPR(pr)
+	statusService := gh.newPRTitleDescriptionStatusService(log, change)
 
 	config := LoadConfiguration(log, change)
-	if gh.HasTitleWithValidType(config, *pullRequest.Title) {
-		return statusService.Success(SuccessMessage, SuccessDetailsPageName)
-	} else if prefix, ok := wip.GetWorkInProgressPrefix(*pullRequest.Title, wip.LoadConfiguration(log, change)); ok {
-		trimmedTitle := strings.TrimPrefix(*pullRequest.Title, prefix)
-		if gh.HasTitleWithValidType(config, trimmedTitle) {
-			return statusService.Success(SuccessMessage, SuccessDetailsPageName)
+	isTitleWithValidType := gh.HasTitleWithValidType(config, *pr.Title)
+	if !isTitleWithValidType {
+		if prefix, ok := wip.GetWorkInProgressPrefix(*pr.Title, wip.LoadConfiguration(log, change)); ok {
```

Comment:
What if the particular repository doesn't use WIP plugin and thus considers such prefix as invalid overall? Just a general observation to think through as with this we are coupling both plugins. And I don't really know yet how to address this use-case, so let's think it through.

/cc @MatousJobanek

Reply:
@bartoszmajsak, We are checking first if PR title has required valid type, if not then we are looking for `WIP` in the title, if yes then remove `WIP` prefix and again check if it has required valid type in PR title

Reply:
This is exactly what the code does, but that was not what I was wondering about. My question is what if we don't use WIP plugin on a given repo, so if you prefix it with "WIP" it should be considered not correct by this plugin then, right?

In addition it couples both plugins together.

Reply:
> My question is what if we don't use WIP plugin on a given repo, so if you prefix it with "WIP" it should be considered not correct by this plugin then, right?

Personally, if the title starts with WIP but the rest is a valid semantic title, then I'd like to have the status check green. However, it would be good to have this note in the documentation.

> In addition it couples both plugins together.

You don't have to see this as a coupling of two plugins but as an additional functionality that is brought to this pluing from another source :-)

If you don't like this state, then we can read the `plugins.yaml` file, check if the repository is using work-in-progress plugin and then activate/deactivate this functionality accordingly. That shouldn't be so hard, I just don't see it necessary - but that is my point of view, I don't insist on it ;-)